_rules/SC2-4-6-descriptive-headings.md

```diff
@@ -0,0 +1,170 @@
+---
+name: Heading is descriptive
+test_type: atomic
+
+description: |
+   This rule checks that headings describe the topic or purpose of the content.
+   
+success_criterion:
+- 2.4.6 # Headings and labels
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Dagfinn Rømen
+- Geir Sindre Fossøy
+- Carlos Duarte
+---
+
+## Test procedure
+
+### Applicability
+
+This rule applies to any element with the [semantic role](#semantic-role) of heading that is either [visible](#visible) or [included in the accessibility tree](#included-in-the-accessibility-tree).
+
+**Note**: This rule only applies to elements with the [semantic role](#semantic-role) of heading. Thus, it is a partial check for WCAG 2.0 success criterion 2.4.6, which applies to all headings. "Heading" is used in its general sense and includes titles and other ways to add a heading to different types of content.
+
+### Expectation
+
+Each target element describes the topic or purpose of the subsequent section.
+
+**Note**: Heading content defines the header of a section (whether explicitly marked up using sectioning content elements, or implied by the heading content itself). A section is a self-contained portion of written content that deals with one or more related topics or thoughts. A section may consist of one or more paragraphs and include graphics, tables, lists and sub-sections.
```

Comment:
I am thinking that the note actually needs to be split out into a definition of "section"

**Definition of "Section"** (input for content, not a finished definition)
A distinct part or subdivision of a document.
A section be defined in many different ways, and with combinations thereof, e.g.: 
- it can explicitly be marked up using WAI-ARIA landmarks or HTML5 sectioning content content
- It can be defined via headings. A heading defines the beginning of a section, and the section ends if a heading on the same or higher level than the level indicating the beginning of the section appears.
A section may consist of one or more paragraphs and include graphics, tables, lists and sub-sections.

**Is this a different requirement?** 
"The subsequent section is a self-contained portion of content that deals with one or more related topics or thoughts." 
It doesn't really fit into the descriptive headings requirement, as I see it.

Reply:
- I would not use the term "section" alone, to avoid confusion with the <section> element. Instead, I would use the phrase "sections of the content" or "sections of the page" used in Technique G130
- I would also not use the term "subsequent" because it may not be subsequent, both visually or in the code - for example, if content is inserted when a heading is pressed.
- Maybe a definition for "section of the content" is "one or more consecutive [HTML] elements that together serve a purpose". I also like the "distinct part or subdivision" but not sure about the rest of the definition.

Reply:
I agree with the use of "section of the content" in stead of section alone. It would fit both the test rule and a definition.

As for the term "subsequent", we made an earlier attempt with "following". Maybe "ensuing" or just "its" could be used? According to G130, "each heading identifies its section of the content".

**Suggestions:**
**Expectation:** "Each target element describes the topic or purpose of its section of the content."

**Definition:** Section of the content
A distinct part or subdivision of a document.

A section of content may consist of one or more paragraphs and include graphics, tables, lists and sub-sections that together serve a purpose.

A section of the content may be defined in different ways, and combinations of these, such as:

- HTML markup, using WAI-ARIA landmarks or HTML5 sectioning elements.
- A heading that defines the beginning of a section of the content.

Reply:
@annethyme If we move "Section of the content" into a definition, we could delete the note. The bit with "The subsequent section is a self-contained portion of content that deals with one or more related topics or thoughts" becomes superfluous. Neither is this part of the SC, but merely a frame for which to assess headings.

Reply:
@DagfinnRomen I like the rephrasing and the definition. I would also support moving the note to a definition for "Section of the content"

Reply:
I have updated the expectation and removed the note. A definition for Section of the content is pending.

Reply:
Definition added.